### ISSUE TITLE
How to identify a specific textfield when using component?

### ISSUE THREAD

**Issue by htsnet** [FlutterFlow/flutterflow-issues#2909]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I am trying to set the value of a textfield that is from a component (TxfSellingPriceArray in red).

The issue is that the set is changing the first field (TxfName in green), not the correct one.

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/2826b723-728e-42ea-a4c4-a9ec958d1d10)

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/4771b872-4fc0-4bb2-8038-27915d7ab197)

The name (green) was changed, not the Selling price (red).

To confirm that, I duplicated the component and gave it another name.

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/7d256e72-556b-44d8-862d-1f57e0324fe4)

And then, the correct field was updated:

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/0bec0be2-9da7-43c2-9933-efc1840e5e46)

When I am creating the action to set form, the list is showing this way:

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/b8d6bd5e-d615-49a9-a0b5-17122f050d51)

I have one name for each instance of the component (first name on the list), but the second name is always the same, because it is the name in the component.

How to identify a specific textfield when using component that I named on tree?

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/3420a6f3-bf48-4e36-a35b-93ba651725e9)

### Expected Behavior

Update the correct field from the tree

### Steps to Reproduce

1) create a component with a textfield
2) create a page and insert this component several times
3) create an action to update one of the middle (not the first) textfield.

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFfhefq0It2sdtE1oSIK/palmMXOU8cTeUzlO1hdwwbI+KrE7MXdePsXBFoWOqHd2NIDlq0nmoGzPToiOHDM/ZdHEyBRoBT1rhxUxDhcEmXRJCKCLjtQEdQTMFPJ2LH57aZmSJQAMFsR3w83XGqJeq+b3qCf9qOYwx5e6fDbOY=

### Context

I can´t change the correct value of my fields.

### Visual documentation

Above

### Environment

```markdown
- FlutterFlow version: 4.1.52+
- Platform: Windows
- Browser name and version: Chrome Versão 124.0.6367.207 (Versão oficial) 64 bits
- Operating system and version affected: Windows 10
```

### General

- [ ] I was collaborating with other users in the same project prior to data loss
- [ ] I navigated between different project views leading up to data loss
- [ ] The UI was running very slowly prior to data loss
- [ ] A crash or other bug occurred prior to data loss
- [ ] Data was not saved in a snapshot
- [ ] I was using flutterflow from multiple devices
- [ ] I was using flutterflow from multiple platforms (e.g. both web and MacOS Desktop app)

### Relative to the time the changes were made, data was lost within

- [ ] 1 minute
- [ ] 1 hour
- [ ] 1 day
- [ ] 1 week

### When following my steps to reproduce, data loss happens

- [X] ~100% of the time
- [ ] ~50% of the time
- [ ] ~10% of the time
- [ ] ~1% of the time
- [ ] Very rarely or only encountered once

**Comment by ignalauret** [comment]
Hey @htsnet thanks for your report. I'm trying to replicate your issue, but I created a component with a textfield as you said, placed three of them on a column, and then added a button with an action that, as you can see, has the three text fields correctly differentiated. 

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/47f52124-3189-463f-9023-ed686eec5d81/1b389f2b-c513-414a-8668-39ca79d5515b?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS80N2Y1MjEyNC0zMTg5LTQ2M2YtOTAyMy1lZDY4NmVlYzVkODEvMWIzODlmMmItYzUxMy00MTRhLTg2NjgtMzljYTc5ZDU1MTViIiwiaWF0IjoxNzE1NzM5MTE1LCJleHAiOjMzMjg2Mjk5MTE1fQ.byECGjCuR2PNfIa5jhKeMf8zGB1vj0uGrckfi8k5pX8 " alt="Screenshot 2024-05-14 at 11.06.55 PM.png" width="350" height="337" />

<br>Could you please create a blank project with the setup that you are using to get this bug, so I can check it out? Please send me the Bug report code when you have it ready. Thanks, and have a great day!

**Comment by htsnet** [comment]
Hey @ignalauret,

Thanks for the quick response. You are right. I created this same situation from scratch and got the same result (estudo-componente-text-field-4b92ok).

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/8e303755-0006-44c8-9356-29ddee6258db)

Then, thinking about my tree, I complicated a little this test tree:

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/8cb0054a-f3b8-4d08-861d-42ed8fe3d25b)

After that, the reference in SET action changed (now, all is -> TextField_1):

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/af6cfae7-bfc4-41c0-950f-d24ec7cff3a8)

but it is still working accordingly.

My official tree is a bit more complex:

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/9755885/87bc0b48-17c1-4522-bcd4-eeb714532305)

I simulated this on this test app, and it is still working.

I will do more tests to understand what is going on on my app (s-v-demo-6avwlq). 

Thanks.
